x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/Expression.java

```diff
@@ -101,7 +102,10 @@ public Object fold(FoldContext ctx) {
 
     public abstract Nullability nullable();
 
-    // the references/inputs/leaves of the expression tree
+    /**
+     * {@link Set} of {@link Attribute}s referenced by this {@link Expression}.
+     * Callers <strong>must</strong> not mutate the result.
+     */
```

Comment:
I believe `AttributeSet` is immutable, isn't it?

Reply:
Ah. It is. No wonder I couldn't see anyone mutate it.

Reply:
It it. I wasn't reading closely enough.